types/inet_test.go

```diff
@@ -58,7 +60,7 @@ func TestInetBuilder(t *testing.T) {
 	require.NoError(t, err)
 
 	a = b.NewArray()
-	require.Equal(t, `["192.168.0.0/24" (null) "192.168.0.0/25" (null) "192.168.0.0/26" "192.168.0.0/27"]`, a.String())
+	require.Equal(t, `["192.168.0.0/24" (null) "192.168.0.0/25" (null) "192.168.0.0/26" "192.168.0.0/27" "192.168.0.1/24"]`, a.String())
```

Comment:
Based on the mask/cidr `192.168.0.1/24` ~ `192.168.0.0/24` ~ `192.168.0.255/24`
Do we really need this?

Reply:
If we don't do this the following test case fails:
https://github.com/cloudquery/cloudquery/pull/11719/files#r1239556847

I think it has value saving the original IP address the inet is based on, even if the values could be considered equal.
Another option is to change the `scalar` logic to ignore the IP address, see https://github.com/cloudquery/plugin-sdk/blob/bc9e6e1ae1bff7a6f2c2699a8a113323d689f1a5/scalar/inet.go#L85

But we should be consistent in both cases (either set it, or ignore it)

Reply:
IMO we need to verify only that single IP `192.168.0.1` is stored and obtained correctly, and then leave the code here as is

Reply:
> IMO we need to verify only that single IP 192.168.0.1 is stored and obtained correctly, and then leave the code here as is

Can you please add more context? Do you mean verify in the plugin tests? We have an inconsistency with how we handle the result of `ParseCIDR` between our `inet` arrow type and the scalar `inet` type. We could drop the usage of `scalar` and only use arrow types maybe to align the logic.
Also `inet` can store both an IP and an subnet, so I'm not sure what you mean by `a single IP`, see https://www.postgresql.org/docs/current/datatype-net-types.html

Reply:
I'm fine with any solution, an, if the inet can also hold the mask, then let's add this fix